docs/release-process.md

```diff
@@ -0,0 +1,249 @@
+
+   You can copy the checklist above into the PR description in the next step.
+
+4. Submit your PR to the `release-*` branch that was created earlier and wait for
+   it to merge.
+
+5. Update the following files in the `main` branch with the same versions as in the
+   steps above, submit them in a PR to `main`, and wait for that PR to be merged:
+
+   - [ ] `docs/component-versions.md`
+   - [ ] `docs/quickstart.md`
+   - [ ] `docs/install/install-script.md`
+   - [ ] `scripts/setup_user_namespaces.sh`
+
+## Generate Release Artifacts and Publish the Release
+
+1. Generate the release manifests on the `release-*` branch:
+
+   ```Shell
+   kustomize build config/default > modelmesh.yaml
+   kustomize build config/runtimes --load-restrictor LoadRestrictionsNone > modelmesh-runtimes.yaml
+   cp config/dependencies/quickstart.yaml modelmesh-quickstart-dependencies.yaml
+   ```
+
+   If you see `Error: unknown flag: --load-restrictor` upgrade your `kustomize` version to 4.x.
+
+2. Generate config archive on the `release-*` branch. The scriptlet below automatically
+   determines the release version and chooses the version of the `tar` command for
+   either Linux or macOS. Verify the correct release `VERSION` was found.
+
+   ```Shell
+   VERSION=$( grep -o -E "newTag: .*$" config/manager/kustomization.yaml | sed 's/newTag: //' )
+   TAR_FILE="config-${VERSION}.tar.gz"
+
+   echo "Release: ${VERSION}"
+
+   if $(tar --version | grep -q 'bsd'); then
+     tar -zcvf ${TAR_FILE} -s /config/config-${VERSION}/ config/;
+   else
+     tar -zcvf ${TAR_FILE} config/ --transform s/config/config-${VERSION}/;
+   fi
+   ```
+
+3. Create a new tag on the `release-*` branch and push it to GitHub using the commands
+   below, or, create a new tag in the next step using the GitHub UI. The new
+   `kserve/modelmesh-controller` image will be published via GitHub Actions.
+
+   ```Shell
+   git tag $VERSION
+   git push upstream $VERSION
+
+   echo https://github.com/kserve/modelmesh-serving/releases/new?tag=${VERSION}
+   ```
+
+4. Create the new release in the GitHub UI from the `release-*` branch (or from the
+   tag created in the previous step). Enter the release tag value (e.g. `v0.10.0`) in
+   the "Release title" field and upload the generated installation manifests ("Release assets")
+   in the "Attach binaries ..." section. Click the "Generate release notes" button which
+   will generate the release description.
+
+   **Note**, if you generated a pre-release (e.g. `v0.10.0-rc0`) then copy the release
+   notes from that and remove them from the pre-release description and revise accordingly.
+
+   https://github.com/kserve/modelmesh-serving/releases/new
+
+5. Compare the release and release artifacts to those of previous releases to make
+   sure nothing was missed.
+
+6. Once the release as been published (a new tag has been pushed), verify the check
+   results by clicking on the check mark (✓) next to the latest commit on the `release-*`
+   branch.
```

Comment:
Maybe it's a silly question, but does this check refer to something like [#308](https://github.com/kserve/modelmesh-serving/actions/runs/3898837488/workflow)? If so, where are FVTs verified if at all?

Reply:
The FVTs are run on Pull Requests only for code changes (i.e. doc changes don't trigger it)

https://github.com/kserve/modelmesh-serving/blob/release-0.10/.github/workflows/run-fvt.yml#L4

So the last actual code change on the `release-*` branch would have made sure the code got tested.